octodns/provider/googlecloud.py

```diff
@@ -0,0 +1,343 @@
+            yield gcloud_record
+        # This is to get results which may be on a "paged" page.
+        # (if more than max_results) entries.
+        if gcloud_iterator.next_page_token:
+            for gcloud_record in self._get_gcloud_records(
+                    gcloud_zone, gcloud_iterator.next_page_token):
+                # yield from is in python 3 only.
+                yield gcloud_record
+
+    def _get_gcloud_zone(self, dns_name, page_token=None, create=False):
+        """Return the ManagedZone which has has the matching dns_name, or
+            None if no such zone exist, unless create=True, then create a new
+            one and return it.
+
+            :param dns_name: fqdn of dns name for zone to get.
+            :type dns_name: str
+            :param page_token: page token for the page to get
+            :type page_token: str
+            :param create: if true, create ManagedZone if it does not exist
+                           already
+
+            :type return: new google.cloud.dns.ManagedZone
+        """
+        # Find the google name for the incoming zone
+        gcloud_zones = self.gcloud_client.list_zones(page_token=page_token)
+        for gcloud_zone in gcloud_zones:
+            if gcloud_zone.dns_name == dns_name:
+                return gcloud_zone
+        else:
+            # Zone not found. Check if there are more results which could be
+            # retrieved by checking "next_page_token".
+            if gcloud_zones.next_page_token:
+                return self._get_gcloud_zone(dns_name,
+                                             gcloud_zones.next_page_token)
+            else:
+                # Nothing found, either return None or else create zone and
+                # return that one (if create=True)
+                self.log.debug('_get_gcloud_zone: zone name=%s, '
+                               'was not found by %s.',
+                               dns_name, self.gcloud_client)
+                if create:
+                    return self._create_gcloud_zone(dns_name)
+
+    def populate(self, zone, target=False, lenient=False):
+        """Required function of manager.py to collect records from zone.
+
+            :param zone: A dns zone
+            :type  zone: octodns.zone.Zone
+            :param target: Unused.
+            :type  target: bool
+            :param lenient: Unused. Check octodns.manager for usage.
+            :type  lenient: bool
+
+            :type return: void
+        """
+
+        self.log.debug('populate: name=%s, target=%s, lenient=%s', zone.name,
+                       target, lenient)
+        before = len(zone.records)
+
+        gcloud_zone = self._get_gcloud_zone(zone.name)
+
+        _records = set()
+        if gcloud_zone:
+            for gcloud_record in self._get_gcloud_records(gcloud_zone):
+                if gcloud_record.record_type.upper() in self.SUPPORTS:
+                    _records.add(gcloud_record)
```

Comment:
If the set isn't being used to de-dup or something you can probably just change this to `not in` and `continue` to skip the record and otherwise get rid of `_records` and the second for each on the line below.

Reply:
yes you are right I will do that.